src/DisputeFactory.ts

```diff
@@ -0,0 +1,110 @@
+        const onChainSlashedParticipants =
+            await this.scmContract.getOnChainSlashedParticipants(
+                joinChannel.channelId
+            );
+        const onChainLatestJoinChannelBlockHash =
+            await this.scmContract.getOnChainLatestJoinChannelBlockHash(
+                joinChannel.channelId
+            );
+
+        const disputeIndex = await this.scmContract.getDisputeLength(
+            joinChannel.channelId
+        );
+
+        // Create and return the complete dispute structure
+        const dispute: DisputeStruct = {
+            channelId: joinChannel.channelId,
+
+            genesisStateSnapshotHash: ethers.keccak256(
+                Codec.encode(
+                    latestOnChainStateSnapshot.stateSnapshot,
+                    Type.StateSnapshot
+                )
+            ),
+
+            latestStateSnapshotHash: ethers.keccak256(
+                Codec.encode(latestStateSnapshot, Type.StateSnapshot)
+            ),
+
+            // TODO: NEEDS IMPLEMENTATION - Should provide finality proof of the latest state
+
+            stateProof: {
+                forkProof: {
+                    forkMilestoneProofs: [] // on chain snapshot is already on the lastest fork
+                },
+                signedBlocks: [] // TODO: Add any additional blocks needed to reach latestStateSnapshot
+            },
+
+            fraudProofs: [],
+
+            onChainSlashes: onChainSlashedParticipants,
+
+            onChainLatestJoinChannelBlockHash:
+                onChainLatestJoinChannelBlockHash || ethers.ZeroHash,
+
+            // TODO:
+            // This commits to the genesis state of the new fork with expanded participants
+            // Should include: new participants, updated balances, incremented forkCnt
+            outputStateSnapshotHash: ethers.keccak256(
+                Codec.encode(latestStateSnapshot, Type.StateSnapshot) // TODO: Should be NEW snapshot with joined participants
+            ),
+
+            exitChannelBlocks: [],
+
+            // TODO: NEEDS CALCULATION - Hash of all data needed for on-chain auditing
+            // Should include: genesisStateSnapshot, latestStateSnapshot, milestoneSnapshots,
+            // joinChannelBlocks, and any other data needed for dispute verification
+            disputeAuditingDataHash: ethers.ZeroHash, // TODO: Calculate actual auditing data hash
+
+            disputer: joinChannel.participant,
+
+            disputeIndex: disputeIndex,
+
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+
+            timeout: {
+                participant: ethers.ZeroAddress,
+                blockHeight: 0,
+                minTimeStamp: 0,
+                forkCnt: 0,
+                isForced: false,
+                previousBlockProducer: ethers.ZeroAddress,
+                previousBlockProducerPostedCalldata: false
+            },
+
+            selfRemoval: false
+        };
+
+        return dispute;
```

Comment:
We should have a single function for populating/creating disputes - it's always a function of your 'state'/perception (let's call this dispute related state `evidence`) and there shouldn't be any special cases - if you have a valid joinChannel threshold, at creation it fetches it and populates the appropriate fields to fully reflect what your opinion is on what a `dispute` should be - as you exchange your opinion/evidence with others and as others exchange their opinion/evidence with you -> you all should converge on the same 'evidence state' (state replication) and reduce the same dispute